### ISSUE TITLE
"Second Language" control has hard-coded English description and does not support `?allowLinks`

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/number-suite-common#35]
In NumberSuiteCommonPreferencesNode.ts:

```typescript
    const loadAllHtmlText = new RichText(
      'To display a second language, run the <a href="{{url}}">“all” version</a> of Number Play.', {
        font: new PhetFont( 12 ),
        links: { url: 'https://phet.colorado.edu/sims/html/number-play/latest/number-play_all.html' },
        visible: false
      } );
```

This should certainly be addressed before publication -- we don't hard-code English strings in production releases.

It's also problematic that this does not support the `allowLinks` query parameter.  This is the responsibility of the sim. See doc in initialize-globals.js, and example in molecule-polarity UnderDevelopmentPanel.ts.

**Comment by pixelzoom** [comment]
There's a usability problem here too.  This text appears in the Preferences dialog, see below.  I seriously doubt that a user of this sim is going to have any idea what `the "all" version` means.  So in addition to addressing the issues identified above, some wordsmithing is needed here.

<img width="467" alt="screenshot_2170" src="https://user-images.githubusercontent.com/3046552/213762772-5b300598-2a2c-4e54-8df6-e99e1960bdc4.png">

How about something like this?

<img width="469" alt="screenshot_2168" src="https://user-images.githubusercontent.com/3046552/213749570-075ac767-d53a-4579-a55a-a5766f4e18b7.png">

**Comment by pixelzoom** [comment]
How about something like this?

<img width="472" alt="screenshot_2167" src="https://user-images.githubusercontent.com/3046552/213749255-fdff40a2-1f6c-442a-a8e5-b82b011399b5.png">

**Comment by pixelzoom** [comment]
Also note that if I run the sim with `?allowLinks=false`, here's what I currently see in Preferences.   Is this what is expected by common code, or is this a common-code bug?

<img width="479" alt="screenshot_2169" src="https://user-images.githubusercontent.com/3046552/213762952-26a60f08-1516-4307-923d-8366d14a9fe5.png">

**Comment by pixelzoom** [comment]
My feeling is that if `allowLinks=false`, we should just omit the entire "To display ..." line.    Trying to cobble together 2 different versions of that line based on the value of `allowLinksProperty` will require multiple translated string fragments, so that we can derive that line of text with and without a hyperlink.

I'm also wondering whether we should hide the entire "Second Language" control when we're not running a multilingual html file.  What value is the user getting out of a control that is disabled, even with the instruction to run the multilingual version?

**Comment by pixelzoom** [comment]
In the above commits, I went ahead and addressed this in a way that I think resolves the issue, simplifies the implementation, and improves the UX.   @chrisklus please review.

Two key changes in the user interface:

* The URL is moved to its own line and is (conditionally) hyperlinked.  This eliminates the need to create multiple translated strings to support `?allowLinks`.
* `"all" version of Number Play` is changed to `multilingual version of this simulation`.  This is more informative, and addresses a bug -- this control is also present in Number Compare.  And since this is part of NumberSuiteCommonPreferencesNode, it may be present in other future sims in the suite.

Key things to review in NumberSuiteCommonPreferencesNode.ts:

* `urlStringProperty` and `urlText` are responsible for dynamically updating the URL.
* `additionalDescription` is responsible for visibility of both `toDisplayASecondLanguageText` and `urlText`, based on whether the control is enabled. 

With `?allowLinks=true` (the default), it looks like this. 

<img width="482" alt="screenshot_2176" src="https://user-images.githubusercontent.com/3046552/213930946-35bdb1cb-2d1e-40d5-89de-08b3a0e7d991.png">

With `?allowLinks=false`, it looks like this. The URL is still shown, but is not hyperlinked.

<img width="479" alt="screenshot_2177" src="https://user-images.githubusercontent.com/3046552/213931014-50134a55-e25c-4738-91d9-81aba51b5a17.png">

With `?locales=en,es` (mulitple locales ), the additional description is hidden:

<img width="472" alt="screenshot_2178" src="https://user-images.githubusercontent.com/3046552/213931025-27872314-f287-444c-9ce8-71f3618d4a96.png">

<img width="487" alt="screenshot_2179" src="https://user-images.githubusercontent.com/3046552/213931029-dd451820-5245-4c4b-bcd7-ef81709c6884.png">

**Comment by pixelzoom** [comment]
A note after refactoring that was done for https://github.com/phetsims/number-suite-common/issues/36... 

@chrisklus you'll now be reviewing SecondLanguageControl.ts.  Inner class `AdditionalDescriptionNode` handles the "To display..." description and URL.

**Comment by chrisklus** [comment]
Thanks @pixelzoom! Code changes look great and behavior is working as expected when testing with and without `allowLinks=false`.

One thing to note regarding this line from https://github.com/phetsims/number-suite-common/issues/35#issuecomment-1399558033:

> `"all" version of Number Play` is changed to `multilingual version of this simulation`.  This is more informative, and addresses a bug -- this control is also present in Number Compare.  And since this is part of NumberSuiteCommonPreferencesNode, it may be present in other future sims in the suite.

I like this language change and noticed that the link in Number Compare was still pointing to Number Play, so @marlitas and I parameterized the `allUrl` in the commit above. We tested and all is working as expected. I think we are good to close here.
